### ISSUE TITLE
Integer Division returns Foreign object

### ISSUE THREAD

**Issue by fniephaus** [oracle/truffleruby#1581]
In Java, `Long.MIN_VALUE / -1` is the only division that overflows. I don't know why, but `-9223372036854775808 / -1` returns a `Foreign` object in TruffleRuby.

truffleruby 1.0.0 RC12 repro session:
```bash
$ truffleruby
[ruby] WARNING: truffleruby starts IRB when stdin is a TTY instead of reading from stdin, use '-' to read from stdin
irb(main):001:0> -9223372036854775808
=> -9223372036854775808
irb(main):002:0> -9223372036854775808 / -1
=> #<Truffle::Interop::Foreign:0x80000000>
```

Related: https://github.com/smarr/SOMns/pull/292.

**Comment by eregon** [comment]
We already handle this case, but by mistake return a `BigInteger` and not a Ruby Bignum/Integer, good catch.

**Comment by fniephaus** [comment]
For the record, these are lines @eregon was referring to: [IntegerNodes L326](https://github.com/oracle/truffleruby/blob/fb14db44e98da85b66bfd4ff60b5426a4efcba46/src/main/java/org/truffleruby/core/numeric/IntegerNodes.java#L326) and [IntegerNodes L367](https://github.com/oracle/truffleruby/blob/fb14db44e98da85b66bfd4ff60b5426a4efcba46/src/main/java/org/truffleruby/core/numeric/IntegerNodes.java#L367).

**Comment by eregon** [comment]
Fixed in 3650346e173a21ac8accbcd4a980545cb3ea2007, thank you for the bug report!

**Comment by fniephaus** [comment]
Sure thing, thanks for the fix!
